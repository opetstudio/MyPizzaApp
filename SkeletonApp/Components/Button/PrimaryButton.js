import React, {Component} from 'react'
import {StyleSheet, Text, TouchableHighlight} from 'react-native'
import LinearGradient from 'react-native-linear-gradient'
import {Colors, Fonts, Metrics} from '../../Themes'
import PropTypes from 'prop-types'

export default class PrimaryButton extends Component {
  static propTypes = {
    onPress: PropTypes.func,
    title: PropTypes.string,
    children: PropTypes.string,
    navigator: PropTypes.object,
    colors: PropTypes.string
  }

  constructor (props) {
    super(props)
    let gradientColor = ['#647BFF', '#0BCDFF']
    this.state = {
      bgColor: this.props.colors === 'gradient' ? gradientColor : this.props.colors === undefined ? ['#fff', '#fff'] : [this.props.colors, this.props.colors],
      txtStyle: this.props.colors === 'gradient' ? styles.customBtnTextGradient : styles.customBtnText
    }
  }

  render () {
    return (
      <TouchableHighlight onPress={this.props.onPress} underlayColor={'transparent'}>
        <LinearGradient colors={this.state.bgColor} style={styles.customBtnBG} start={{x: 0, y: 1}} end={{x: 1, y: 1}}>
          <Text style={this.state.txtStyle} >{this.props.title}</Text>
        </LinearGradient>
      </TouchableHighlight>
    )
  }
}

const styles = StyleSheet.create({

  customBtnTextGradient: {
    color: Colors.snow,
    textAlign: 'center',
    fontWeight: 'bold',
    fontSize: Fonts.size.medium,
    marginVertical: Metrics.baseMargin
  },

  customBtnText: {
    color: Colors.blackSecondaryOpacity,
    textAlign: 'center',
    fontWeight: 'bold',
    fontSize: Fonts.size.medium,
    marginVertical: Metrics.baseMargin
  },

  customBtnBG: {
    height: 35,
    borderRadius: 5,
    marginHorizontal: 50,
    marginVertical: Metrics.baseMargin,
    backgroundColor: Colors.fire,
    justifyContent: 'center',
    elevation: 5
  }
})
