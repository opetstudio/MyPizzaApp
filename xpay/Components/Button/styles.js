import { StyleSheet } from 'react-native'
import { Colors as colors } from '../../Themes'

export const styles = StyleSheet.create({
  container: {
    flex: 1
  },
  button: {
    alignItems: 'center',
    paddingVertical: 10
  },
  // for Primary Button other component should be flex: 1
  primary: {
    backgroundColor: colors.redPrimary,
    flex: 1
  },
  primaryInactive: {
    backgroundColor: colors.redPrimaryOpacity,
    flex: 1
  },
  // for Secondary Button other component should be flex: 1
  secondary: {
    backgroundColor: colors.darkNavyBlueSecondary,
    borderColor: colors.redPrimary,
    borderWidth: 1,
    flex: 1
  },
  secondaryInactive: {
    // backgroundColor: colors.darkNavyBlueSecondaryOpacity,
    backgroundColor: 'blue',
    borderColor: colors.redPrimaryOpacity,
    flex: 1
  }
})
