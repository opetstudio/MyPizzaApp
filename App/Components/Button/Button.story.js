import React from 'react'
import { storiesOf } from '@storybook/react-native'
// import {
//   View,
//   Image,
//   TouchableOpacity
// } from 'react-native'
// import StyledText from '../StyledText'
// import {Images} from '../../Themes'

// import Button from './index'
import PrimaryButton from './PrimaryButton'

storiesOf('Button')
  .add('primaryButton', () => (
    // without title
    <PrimaryButton />
  ))
  .add('primaryButton with label', () => (
    // using title
    <PrimaryButton title={'Title'} />
  ))
  .add('primaryButton with gradient background', () => (
    // using title
    <PrimaryButton colors={'gradient'} title={'Gradient background'} />
  ))
  .add('primaryButton with green background', () => (
    // using title
    <PrimaryButton colors={'#0f0'} title={'Green Button'} />
  ))
  .add('primaryButton with red background', () => (
    // using title
    <PrimaryButton colors={'#f00'} title={'Red Button'} />
  ))
  .add('primaryButton with blue background', () => (
    // using title
    <PrimaryButton colors={'#00f'} title={'Blue Button'} />
  ))
  // .add('withImage', () => (
  //   <Button type='tab' onPress={() => { alert('test') }}>
  //     <View style={{backgroundColor: 'red'}}>
  //       <StyledText
  //         i18nKey='dashboard-more'
  //         textStyle='h13LtGreyS'
  //         upperCase={false}
  //     />
  //       <Image
  //         source={Images.chevronRight2}
  //     />
  //     </View>
  //   </Button>
  // ))
  // .add('link', () => (
  //   <Button
  //     onPress={() => {}}
  //     i18nKey={'view more'}
  //     upperCase={false}
  //     type='link'
  //     addedStyle={{backgroundColor: 'red'}}
  //     addedTextStyle={{color: 'blue'}}
  //   />
  // ))
  // .add('tab', () => (
  //   <Button
  //     onPress={() => {}}
  //     i18nKey={'tab menu'}
  //     upperCase={false}
  //     type='tab'
  //     addedStyle={{backgroundColor: 'red'}}
  //     addedTextStyle={{color: 'yellow'}}
  //   />
  // ))
  // .add('primary', () => (
  //   <Button
  //     onPress={() => {}}
  //     i18nKey={'Primary'}
  //     upperCase={false}
  //     addedStyle={{backgroundColor: 'red'}}
  //     addedTextStyle={{color: 'yellow'}}
  //   />
  // ))
  // .add('RowWithImageThums', () => (
  //   <RowList componentHeight={100} textUp='Contact Name'
  //     textDown='1234567890'
  //     Text='Mobile' />
  // ))
