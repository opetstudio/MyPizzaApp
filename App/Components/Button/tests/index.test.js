import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import Button from '../index';

describe('Button', () => {
  const mockOnPress = jest.fn();

  it('should render correctly', () => {
    const tree = shallow(<Button />);
    expect(tree).toMatchSnapshot();
  });

  it('should simulate button click', () => {
    const component = shallow(<Button onPress={mockOnPress} />);
    const ButtonComponent = component.find('StyledTouchableOpacity');
    ButtonComponent.simulate('Press', {
      nativeEvent: { timestamp: Date.now() },
    });
    expect(mockOnPress).toHaveBeenCalledTimes(1);
  });

  it('should render the link type Button', () => {
    const tree = shallow(<Button onPress={mockOnPress} type="link" />);
    expect(tree).toMatchSnapshot();
  });

  it('should render isInactive type Button', () => {
    const tree = shallow(<Button isInactive />);
    expect(tree).toMatchSnapshot();
  });

  it('should render noFeedback type Link', () => {
    const tree = shallow(<Button noFeedback type="link" />);
    expect(tree).toMatchSnapshot();
  });
});
