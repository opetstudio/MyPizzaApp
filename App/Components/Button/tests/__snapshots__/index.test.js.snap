// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render correctly 1`] = `
<StyledTouchableOpacity
  isMultipleTapAllowed={true}
  style={
    Array [
      Object {
        "alignItems": "center",
        "paddingVertical": 10,
      },
      undefined,
      undefined,
    ]
  }
>
  <StyledText
    isUnderline={false}
    letterSpacing={0}
    textStyle="h11MedWhiteT3"
  />
</StyledTouchableOpacity>
`;

exports[`Button should render isInactive type Button 1`] = `
<StyledTouchableOpacity
  disabled={true}
  isMultipleTapAllowed={true}
  style={
    Array [
      Object {
        "alignItems": "center",
        "paddingVertical": 10,
      },
      undefined,
      undefined,
    ]
  }
>
  <StyledText
    isUnderline={false}
    letterSpacing={0}
    textStyle="h11MedWhiteT3O"
  />
</StyledTouchableOpacity>
`;

exports[`Button should render noFeedback type Link 1`] = `
<StyledTouchableOpacity
  activeOpacity={1}
  style={
    Array [
      undefined,
      undefined,
    ]
  }
>
  <StyledText
    addedStyle={
      Array [
        undefined,
        undefined,
      ]
    }
    isUnderline={false}
    letterSpacing={0}
    textStyle="h11MedRedP"
  />
</StyledTouchableOpacity>
`;

exports[`Button should render the link type Button 1`] = `
<StyledTouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Array [
      undefined,
      undefined,
    ]
  }
>
  <StyledText
    addedStyle={
      Array [
        undefined,
        undefined,
      ]
    }
    isUnderline={false}
    letterSpacing={0}
    textStyle="h11MedRedP"
  />
</StyledTouchableOpacity>
`;
